### PR TITLE
Remove LINQ allocations from FAR item display logic

### PR DESCRIPTION
**Pull request by rchande**
This usage of .Select is responsible for 1.3% of allocations if you FAR for SyntaxNode.

**Customer scenario**

This will potentially make find all references faster for all customers.

**Bugs this fixes:** 

https://github.com/dotnet/roslyn/issues/13326

**Workarounds, if any**

N/A, this is a performance improvement.

**Risk**

Extremely low, this replaces a call to .Select with a foreach loop.

**Performance impact**

Positive--allocation of the WhereSelectEnumerator was showing up in performance traces and now we will not allocate it.

**Is this a regression from a previous update?** 
N/A

**Root cause analysis:**

We prefer to use LINQ when possible and remove usages that show up on hot paths.

**How was the bug found?**

Ad hoc testing in response to telemetry showing FAR performance was out of goal.

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide for review.
Part of https://github.com/dotnet/roslyn/issues/13326

File: src/Features/Core/Portable/Common/TaggedText.cs [hunk 1]
```diff
@@ -57,9 +58,22 @@ public static ImmutableArray<TaggedText> ToTaggedText(this IEnumerable<SymbolDis
...
                 ? null
-                : string.Join("", values.Select(t => t.Text));
+                : Join(values);
+        }
+
+        private static string Join(ImmutableArray<TaggedText> values)
+        {
+            var pooled = PooledStringBuilder.GetInstance();
+            var builder = pooled.Builder;
+            for (int i = 0; i < values.Length; i++)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Common/TaggedText.cs]: use foreach. it will be faster (less index checks).

**Comment by CyrusNajmabadi** [comment]
👍 With the change to foreach.

**Comment by rchande** [comment]
@CyrusNajmabadi Done.

**Comment by Pilchie** [comment]
Tagging @mattwar for escrow approval.

**Comment by mattwar** [comment]
You mean tagging @mattgertz for escrow approval.

**Comment by MattGertz** [comment]
I think he meant Matt Dillon from "Gunsmoke."

Regardless, let's take it.

**Comment by natidea** [comment]
Approved to merge via [370343](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/370343)

**Pull request merged**